app/src/main/java/ar/com/wolox/android/cookbook/home/HomePresenter.kt

```diff
@@ -0,0 +1,14 @@
+package ar.com.wolox.android.cookbook.home
+
+import ar.com.wolox.android.cookbook.utils.SharedPreferencesUtils
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import javax.inject.Inject
+
+class HomePresenter @Inject constructor(
+    private val sharedPreferencesUtils: SharedPreferencesUtils
+) : BasePresenter<HomeView>() {
+
+    fun onContinueButtonClicked() {
+        if (sharedPreferencesUtils.hasReadInfo) view?.goToRecipePickerScreen() else view?.goToInfoScreen()
```

Comment:
view can be null? i think it goes against MVP architecture but im not sure

Reply:
This is not the case, i will change it. Nonetheless, It could be null, for example, if background event has finished when user has changed screen.